controller/controller.go

```diff
@@ -141,6 +145,9 @@ func (r Request) ExtendSlices() (Request, error) {
 }
 
 func (c controller) Submit(req Request) (*task.TaskObject, error) {
+	if (len(req.Units) == 0) {
+		return nil, errgo.WithCausef(nil, maskAny(invalidArgumentError) , "invalid argument: Units must not be empty")
```

Comment:
Here we see why I added `maskAnyf` in my branch. At first it is more convenient. Then it automatically combines the error message of the cause with the additional format string.